File: 2st/Riyeon/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Stack<Character> st = new Stack<>();
        for(char ch:s.toCharArray()){
            if(ch=='(')
                st.push(ch);
            else if(ch==')'){
                if(st.empty())
                    return false;
                st.pop();
            }
        }
        return st.empty();
    }
}

File: 4st/Hani/programmer/올바른 괄호.js
function solution(s) {
  var answer = true;
  let stack = [];

  for (let i of s) {
    if (i === "(") {
      stack.push(i);
    } else if (i === ")") {
      if (stack.length === 0) {
        answer = false;
        return answer;
      }
      stack.pop(i);
    }
  }
  if (stack.length === 0) {
    return answer;
  } else {
    answer = false;
    return answer;
  }
}
